pkg/apiserver/watch.go

```diff
@@ -196,7 +197,9 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 				// client disconnect.
 				return
 			}
-			flusher.Flush()
+			if len(ch) == 0 {
```

Comment:
I'm not convinced about this change. If the chrun in the cluster is big enough, ch can potentially never be empty.

I think we should flush no matter what at least 1 per X milliseconds.